src/ScraperType/CourseTopicScraper/ScraperModules/ApiUtility.py

```diff
@@ -76,19 +76,32 @@ def getCourseCollectionsJson(self, courseApiUrl, courseUrl):
             topicNameList = []
             baseApiUrl = f"https://educative.io/api/collection/{authorId}/{collectionId}/page/"
             categoryType = ["COLLECTION_PROJECT", "COLLECTION_CATEGORY", "COLLECTION_ASSESSMENT", "PATH_EXTERNAL_PROJECT", "PATH_EXTERNAL_ASSESSMENT", "CLOUD_LAB"]
+            idx = 0
+            toc = []
             for category in categories:
                 if any(cType in category["type"] for cType in categoryType) and (
                         isinstance(category["id"], int) or len(category["id"]) <= 10):
                     if not category["pages"]:
                         topicApiUrlList.append(baseApiUrl + str(category["id"]) + f"?work_type={courseType}")
                         topicNameList.append(category["title"])
+                        category_topic = (idx, category["title"], baseApiUrl + str(category["id"]) + f"?work_type={courseType}")
+                        toc.append(category_topic)
+                        idx += 1
+                    else:
+                        category_topic = {"category": category["title"], "topics": []}
+                        toc.append(category_topic)
+
                     for page in category["pages"]:
                         topicApiUrlList.append(baseApiUrl + str(page["id"]) + f"?work_type={courseType}")
```

Comment:
@anhpho same for here, create a variable and store the api url and use it later on